src/main/java/exceptions/InvalidSyntaxException.java

```diff
@@ -0,0 +1,15 @@
+package exceptions;
+
+public class InvalidSyntaxException extends Exception {
+
+    private final String expectedSyntax;
```

Comment:
If this string is not a constant, u should remove the "final". If it is indeed a constant, the name should be all uppercase using underscore to separate words. Similar issue in other places.

Reply:
Thanks for the comment, but I think this is alright -- `expectedSyntax` here is a blank final variable which can, and must, only be initialized in the constructor of `InvalidSyntaxException`. I think together with the private access modifier, this makes it clear that each instance of `InvalidSyntaxException` is associated with a specific `expectedSyntax`, and that it must not be modified after the constructor